src/ngamsServer/ngamsServer/janitor/rotated_logfiles_handler.py

```diff
@@ -61,19 +74,21 @@ def run(srvObj, stopEvt):
     # ngamsServer.NgasRotatingFileHandler class, which is the file handler
     # attached to the root logger in the server process
     logger.debug("Checking if there are unsaved rotated logfiles")
-    for unsaved in glob.glob(os.path.join(logdir, 'LOG-ROTATE-*.nglog.unsaved')):
+    for unsaved in sorted(glob.glob(os.path.join(logdir, 'LOG-ROTATE-*.nglog.unsaved'))):
 
         # Remove the .unsaved bit, leave the rest
         fname = '.'.join(unsaved.split('.')[:-1])
-        mvFile(unsaved, fname)
-
-        # Connect to the server and send a pull ARCHIVE request
-        if cfg.getArchiveRotatedLogfiles():
-            file_uri = "file://" + fname
-            host, port = srvObj.get_self_endpoint()
-            proto = srvObj.get_server_access_proto()
-            ngamsPClient.ngamsPClient(host, port, proto=proto).archive(
-                    file_uri, 'ngas/nglog')
+        try:
+            mvFile(unsaved, fname)
+            try_archiving(cfg, srvObj, fname)
+        except Exception as e:
+            mvFile(fname, unsaved)
```

Comment:
Based on our conversation, is it necessary to first `mvFile` then on a possible exception restore? Is it more worthwhile to `mvFile` only after the successful archive?

Reply:
Yeah, I absolutely agree that the first move is not required in principle, and would make exception handling simpler. However, the `ngamsPClient.archive` method doesn't allow to pass down arbitrary filenames, and will always use the basename of the file on disk that is being archived.

As discussed, I'll just go ahead then with these changes, but will create an issue to remember implementing this improvement in the future.